src/overlays/actors/ovl_En_Butte/z_en_butte.c

```diff
@@ -21,41 +45,376 @@ const ActorInit En_Butte_InitVars = {
+    ICHAIN_F32(uncullZoneScale, 20, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 600, ICHAIN_STOP),
+};
+
+void EnButte_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnButte* this = THIS;
+
+    if (this->actor.params == -1) {
+        this->actor.params = 0;
+    }
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    if ((this->actor.params & 1) == 1) {
+        this->actor.uncullZoneScale = 200.0f;
+    }
+
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_050036F0, &D_05002470, this->limbDrawTable,
+                   this->transitionDrawTable, 8);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sColliderInit, this->colliderItems);
+    this->actor.colChkInfo.mass = 0;
+    this->unk_25C = Math_Rand_ZeroOne() * 0xFFFF;
+    this->unk_25E = Math_Rand_ZeroOne() * 0xFFFF;
+    this->unk_260 = Math_Rand_ZeroOne() * 0xFFFF;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_05002470, 1.0f, 0.0f, 0.0f, 1, 0.0f);
+    EnButte_SetupFlyAround(this);
+    this->actor.shape.rot.x -= 0x2320;
+    this->drawSkelAnime = true;
+    // "field keep butterfly"
+    osSyncPrintf("(field keep 蝶)(%x)(arg_data 0x%04x)\n", this, this->actor.params);
+}
+
+void EnButte_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    Collider_DestroyJntSph(globalCtx, &THIS->collider);
+}
+
+void func_809CD56C(EnButte* this) {
+    static f32 D_809CE3E0[] = { 50.0f, 80.0f, 100.0f };
+    static f32 D_809CE3EC[] = { 30.0f, 40.0f, 50.0f };
+
+    this->actor.shape.unk_08 += Math_Sins(this->unk_25C) * D_809CE3E0[this->flightParamsIdx] +
+                                Math_Sins(this->unk_25E) * D_809CE3EC[this->flightParamsIdx];
+    this->actor.shape.unk_08 = CLAMP(this->actor.shape.unk_08, -2000.0f, 2000.0f);
+}
+
+void func_809CD634(EnButte* this) {
+    static f32 D_809CE3F8[] = { 15.0f, 20.0f, 25.0f };
+    static f32 D_809CE404[] = { 7.5f, 10.0f, 12.5f };
+
+    this->actor.shape.unk_08 += Math_Sins(this->unk_25C) * D_809CE3F8[this->flightParamsIdx] +
+                                Math_Sins(this->unk_25E) * D_809CE404[this->flightParamsIdx];
+    this->actor.shape.unk_08 = CLAMP(this->actor.shape.unk_08, -500.0f, 500.0f);
+}
+
+void EnButte_Turn(EnButte* this) {
+    s16 target = this->actor.posRot.rot.y + 0x8000;
+    s16 diff = target - this->actor.shape.rot.y;
+
+    Math_ApproxUpdateScaledS(&this->actor.shape.rot.y, target, ABS(diff) >> 3);
+    this->actor.shape.rot.x = (s16)(sinf(this->unk_260) * 600.0f) - 0x2320;
+}
+
+void EnButte_SetupFlyAround(EnButte* this) {
+    EnButte_SelectFlightParams(this, &sFlyAroundParams[this->flightParamsIdx]);
+    this->actionFunc = EnButte_FlyAround;
+}
+
+void EnButte_FlyAround(EnButte* this, GlobalContext* globalCtx) {
+    EnButteFlightParams* flightParams = &sFlyAroundParams[this->flightParamsIdx];
+    s16 yaw;
+    Player* player = PLAYER;
+    f32 distFromHome;
+    f32 maxDistFromHome;
```

Comment:
From what I can tell these are distances squared 
```suggestion
    f32 distSqFromHome;
    f32 maxDistSqFromHome;
```